### PR TITLE
Cherry-pick #20915 to 7.x: [Autodiscover] Handle input-not-finished errors in config reload

### PR DESCRIPTION
**Pull request by ChrsMark**
Cherry-pick of PR #20915 to 7.x branch. Original message: 

## What does this PR do?
This PR adds special error handling in config reload flow to handle state errors specially by not logging in ERROR level to avoid confusion in logging.

Current state:
```
2020-08-26T07:05:05.386Z        ERROR   [autodiscover]  cfgfile/list.go:95      Error creating runner from config: Can only start an input when all related states are finished: {Id: ea745ab688be85a9-native::1308836-2049, Finished: false, Fileinfo: &{628a5ca3a5e5037056ebabc9a016b13d4a6ebb16c44a6216beff387b65ff5c4b-json.log 99 416 {29223485 63734022237 0x608b880} {2049 1308836 1 33184 0 0 0 0 99 4096 8 {1598425436 865214956} {1598425437 29223485} {1598425437 29223485} [0 0 0]}}, Source: /var/lib/docker/containers/628a5ca3a5e5037056ebabc9a016b13d4a6ebb16c44a6216beff387b65ff5c4b/628a5ca3a5e5037056ebabc9a016b13d4a6ebb16c44a6216beff387b65ff5c4b-json.log, Offset: 42864, Timestamp: 2020-08-26 07:05:03.816329242 +0000 UTC m=+187.623544858, TTL: -1ns, Type: container, Meta: map[stream:stdout], FileStateOS: 1308836-2049}
```

Patch:
```
2020-09-02T08:19:12.330Z	DEBUG	[autodiscover]	cfgfile/list.go:97	Error creating runner from config: Can only start an input when all related states are finished: {Id: native::5784858-64769, Finished: false, Fileinfo: &{mytarget2_default_test-63b37f8e982045d4bdac75d302e687e788e979ffeb857b5c1e754f019275d6e5.log 15108 416 {330272804 63734631529 0x36610e0} {64769 5784858 1 33184 0 0 0 0 15108 4096 32 {1599034103 795344632} {1599034729 330272804} {1599034729 330272804} [0 0 0]}}, Source: /var/log/containers/mytarget2_default_test-63b37f8e982045d4bdac75d302e687e788e979ffeb857b5c1e754f019275d6e5.log, Offset: 15465, Timestamp: 2020-09-02 08:19:05.456366051 +0000 UTC m=+12.261964777, TTL: -1ns, Type: container, Meta: map[], FileStateOS: 5784858-64769}
```

## Why is it important?
To avoid confusion in logging because of a non actual ERROR is logged as an ERROR. We need to log with lower severity this kind of errors.

## How to test this PR locally

Using testing steps from https://github.com/elastic/beats/pull/20305.

## Related issues

- Closes https://github.com/elastic/beats/issues/20850
- Relates https://github.com/elastic/beats/pull/20305, https://github.com/elastic/beats/issues/11834, https://github.com/elastic/beats/pull/20898#pullrequestreview-480549739

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -184,6 +184,8 @@ field. You can revert this change by configuring tags for the module and omittin
 - [Metricbeat][Kubernetes] Change cluster_ip field from ip to keyword. {pull}20571[20571]
 - Rename cloud.provider `az` value to `azure` inside the add_cloud_metadata processor. {pull}20689[20689]
 - Add missing country_name geo field in `add_host_metadata` and `add_observer_metadata` processors. {issue}20796[20796] {pull}20811[20811]
+- [Autodiscover] Handle input-not-finished errors in config reload. {pull}20915[20915]
+- Explicitly detect missing variables in autodiscover configuration, log them at the debug level. {issue}20568[20568] {pull}20898[20898]
```
Comment (jsoriano) [inline CHANGELOG.next.asciidoc:188]: This line is not related to this change.

Comment (ChrsMark) [reply CHANGELOG.next.asciidoc]: 😬 thanks!

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -185,7 +185,7 @@ field. You can revert this change by configuring tags for the module and omittin
 - [Metricbeat][Kubernetes] Change cluster_ip field from ip to keyword. {pull}20571[20571]
 - Rename cloud.provider `az` value to `azure` inside the add_cloud_metadata processor. {pull}20689[20689]
 - Add missing country_name geo field in `add_host_metadata` and `add_observer_metadata` processors. {issue}20796[20796] {pull}20811[20811]
-- Explicitly detect missing variables in autodiscover configuration, log them at the debug level. {issue}20568[20568] {pull}20898[20898]
```
Comment (jsoriano) [inline CHANGELOG.next.asciidoc:188]: Umm, but it shouldn't be removed :slightly_smiling_face:

Comment (ChrsMark) [reply CHANGELOG.next.asciidoc:188]: 🤦 messy update/merge, sorry

**Review by jsoriano — APPROVED** [review]
